Release channels (#717)
kyma-project/community: concepts/modularization/README.md

If the API you need (like a core Kubernetes API or Istio virtual service) is not available, you should fail. If your component can work without the API, but some features are not available (for example, service monitor from monitoring), you should just skip it and continue to deploy other component resources. 

# Release channels

> Release channels let customers try new modules or features early or decide when the updates will be applied. 

Reviewer: ```suggestion
Release channels let customers try new modules and features early, and decide when the updates should be applied. 
```

![](assets/release-channels.drawio.svg)

> The first use case will be modeled as the alpha channel. Modules available in the alpha channel are developed with all quality measures in place (functional correctness, security, etc), but can have still unstable API or can be changed without keeping backward compatibility. When you use the module from alpha channel you won't get full SLA guarantees for that module or other modules that are affected (directly or indirectly).

Reviewer: ```suggestion
The first use case will be modeled as the alpha channel. Modules available in the alpha channel are developed with all quality measures in place (functional correctness, security, etc.), but they might still have unstable API or be changed without keeping backward compatibility. When you use a module from the alpha channel, you won't get full SLA guarantees for that module or other modules that are affected (directly or indirectly).
```

> The second use case (decide when updates should be applied) will require 2 production-grade channels with the different update schedule. The fast channel will get updates as soon as they are released and pass all quality gates. The regular channel will get updates a few days later. Customers can switch the entire cluster or particular component to the fast channel to check if the upstream changes will not cause any issues with their workload. Changing back to the regular channel is possible but the module version won't be downgraded - the next version has to reach the channel to be applied.

Reviewer: ```suggestion
The second use case (deciding when updates should be applied) will require 2 production-grade channels with a different update schedule. The fast channel will get updates as soon as they are released and have passed all quality gates. The regular channel will get updates a few days later. Customers can switch the entire cluster or a particular component to the fast channel to check if the upstream changes do not cause any issues with their workload. Changing back to the regular channel is possible, but the module version won't be downgraded - the next version has to reach the channel to be applied.
```

> Hotfixes will be delivered to all channels immediately (TODO: how to apply hotfix for the release that is not available in the current channel).

Reviewer: ```suggestion
Hotfixes will be delivered to all channels immediately (TODO: how to apply a hotfix for the release that is not available in the current channel).
```
